Source/ARTRestChannel.m

```diff
@@ -214,6 +218,58 @@ - (BOOL)history:(ARTDataQuery *)query callback:(ARTPaginatedMessagesCallback)cal
     return ret;
 }
 
+- (void)status:(ARTChannelDetailsCallback)callback {
+    if (callback) {
+        ARTChannelDetailsCallback userCallback = callback;
+        callback = ^(ARTChannelDetails *details, ARTErrorInfo *_Nullable error) {
+            dispatch_async(self->_userQueue, ^{
+                userCallback(details, error);
+            });
+        };
+    }
+    dispatch_async(_queue, ^{
+        NSURL *url = [NSURL URLWithString:self->_basePath];
+        NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:url];
```

Comment:
>I've just noticed we should probably also set the Accept header on the request, [again similarly to -[ARTRest time:]]

I think it's a legacy code. `Accept` header is set later here for all requests:
https://github.com/ably/ably-cocoa/blob/0fe3c59658ce7d0b7181571cdad1f068d4085088/Source/ARTRest.m#L313

Reply:
>If this is the case, please could you make an issue (or update the existing one) so that we address it here too?

I've replied there https://github.com/ably/ably-cocoa/issues/1442#issuecomment-1179390433

Reply:
But there is still an issue with this
```
id<ARTEncoder> encoder = self->_rest.encoders[response.MIMEType];
ARTChannelDetails *channelDetails = [encoder decodeChannelDetails:data error:&decodeError];
```
`channelDetails` can be nil if `encoder` is nil and the callback will be called with both params being `nil`.